src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2205,16 +2205,23 @@ primitive_inst::primitive_inst(network & network, program_node const& node, bool
             }
         }
 
-        // TODO: Remove WA for arg_max_min node.
-        // For now it's required to handle the case when only second output of TopK primitive is used in plugin,
-        // but kernels always write both outputs to the same memory object which leads to wrong result.
-        if (user_count == 1 && mutable_data_count == 1 && !node.is_type<arg_max_min>()
-                                                       && !node.is_type<experimental_detectron_roi_feature_extractor>()) {
-            for (auto& user : node.get_users())
-                if (user->is_type<mutable_data>())
-                    _outputs[0] = user->as<mutable_data>().get_attached_memory_ptr();
+        if (node.is_reusing_memory() ) {      
+                const auto& eltw_inst = _network.get_primitive(node.reuse_id());
```

Comment:
`eltw_inst` is a bit difficult to understand without context.
* `eltw_inst` -> `output_buffer_inst`
* Could you add comment? `// sum post-op can use the input buffer as the output buffer`